src/libcollections/str.rs

```diff
@@ -96,11 +96,11 @@ impl<S: Str> SliceConcatExt<str, String> for [S] {
         }
 
         // `len` calculation may overflow but push_str will check boundaries
-        let len = s.iter().map(|s| s.as_slice().len()).sum();
+        let len = s.iter().map(|s| s.as_str().len()).sum();
```

Comment:
Yeah the slicing syntax should definitely be usable as-is today, thankfully it's been around for awhile!